internal/lsp/diagnostics.go

```diff
@@ -17,3 +17,24 @@ func HCLSeverityToLSP(severity hcl.DiagnosticSeverity) lsp.DiagnosticSeverity {
 	}
 	return sev
 }
+
+func HCLDiagsToLSP(hclDiags hcl.Diagnostics) []lsp.Diagnostic {
+	diags := []lsp.Diagnostic{}
+
+	for _, hclDiag := range hclDiags {
+		// only process diagnostics with an attributable spot in the code
```

Comment:
Why do we omit subject-less diagnostics here? 🤔 

I'm not sure what kind of diagnostics these may be in practice, so it's hard for me to judge the impact, I just remember that in case of Terraform's own diags (e.g. exposed via `terraform validate` or by any command which parses code really) this is more common than we'd like.

For example providers are common source of such diags, because provider blocks are technically optional in the configuration: https://github.com/hashicorp/terraform/issues/20121

Reply:
I don't have any good reason really, I'll see what the UX is like for a non subject diag